File: Data Structures & Algorithms/merge-intervals/submission-4.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        k = intervals[0]
        res = []

        for start, end in intervals[1:]:
            if k[1] >= start:
                k = [min(k[0], start), max(k[1], end)]
            else:
                res.append(k)
                k = [start, end]
        
        res.append(k)
        return res

File: Data Structures & Algorithms/longest-common-prefix/submission-4.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string}
     */
    longestCommonPrefix(strs) {
        strs.sort()
        const first = strs[0]
        const last = strs[strs.length - 1]
        let minn = Math.min(first.length, last.length)

        let res = ''
        for (let i=0; i < minn; i++){
            if (first[i] != last[i]){
                break 
            }
            else {
                res += first[i]
            }
        }

        return res
    }
}
